src/org/labkey/test/pages/dataset/EditDatasetDefinitionPage.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    public EditDatasetDefinitionPage setDescription(String description)
+    {
+        expandPropertiesPanel();
+        elementCache().descriptionInput.set(description);
+        return this;
+    }
+
+    public String getDescription()
+    {
+        return elementCache().descriptionInput.get();
+    }
+
+    // properties panel interactions
+    // get/set category
+    public EditDatasetDefinitionPage setCategory(String category)
+    {
+        expandPropertiesPanel();
+        elementCache().categorySelect.filterSelect(category);
+        return this;
+    }
+
+    public String getCategory()
+    {
+        return elementCache().categorySelect.getValue();
+    }
+
+    // get/set label
+    public EditDatasetDefinitionPage setDatasetLabel(String label)
+    {
+        expandPropertiesPanel();
+        elementCache().labelInput.set(label);
+        return this;
+    }
+
+    public String getDatasetLabel()
+    {
+        return elementCache().labelInput.get();
+    }
+
+    // get/select radio button for: participants only(demographic data)/participants and visit/participants, visit, and additional key field
+    public EditDatasetDefinitionPage setIsDemographicData(boolean checked)
+    {
+        expandPropertiesPanel();
+        elementCache().participantsOnlyRadioBtn.set(checked);
+        return this;
+    }
+
+    public boolean isDemographicsData()
+    {
+        return  elementCache().participantsOnlyRadioBtn.get();
+    }
+
+    public String getAdditionalKeyColDataField()
+    {
+        if (!elementCache().keyFieldSelect.isEnabled())
+            return null;
+        else
+            return elementCache().keyFieldSelect.getValue();
+    }
+
+    public EditDatasetDefinitionPage setAdditionalKeyColDataField(String field)
+    {
+        setAdditionalKeyColumnType(LookupAdditionalKeyColType.MANAGEDFIELD);
+        elementCache().keyFieldSelect.select(field);
+        return this;
+    }
+
+    public EditDatasetDefinitionPage setAdditionalKeyColManagedField(String field)
+    {
+        setAdditionalKeyColumnType(LookupAdditionalKeyColType.MANAGEDFIELD);
+        elementCache().keyFieldSelect.select(field);
+        elementCache().keyPropertyManagedBox.check();
+        return this;
+    }
+
+    public EditDatasetDefinitionPage setAdditionalKeyColumnType(LookupAdditionalKeyColType type)
+    {
+        expandPropertiesPanel();
+        new RadioButton(elementCache().dataRowRadioBtn(type.getIndex()).findElement(getDriver())).check();
+        return this;
+    }
+
+    public boolean isAdditionalKeyManagedEnabled()
+    {
+        expandPropertiesPanel();
+        return elementCache().additionalKeyFieldRadioBtn.isChecked() &&
+                elementCache().keyFieldSelect.hasValue() &&
+                elementCache().keyPropertyManagedBox.isChecked();
+    }
+
+    public boolean isAdditionalKeyDataFieldEnabled()
+    {
+        return  elementCache().keyFieldSelect.isEnabled();
```

Comment:
The method name and the check here don't seem to match. If this is just checking the enabled state of the keyFieldSelect, then the "Data" part of the method name should be dropped.